src/services/store.ts

```diff
@@ -59,13 +60,19 @@ export function addLastUsedDataSourceToStorage(dsKey: string) {
 }
 
 const SORT_BY_LOCALSTORAGE_KEY = `${pluginJson.id}.values.sort`;
-export function getSortByPreference(target: string, defaultSortBy: string, defaultDirection: 'desc' | 'asc') {
+export function getSortByPreference(
+  target: string,
+  defaultSortBy: SortBy,
+  defaultDirection: SortDirection
+): { sortBy: SortBy | ''; direction: SortDirection } {
   const preference = localStorage.getItem(`${SORT_BY_LOCALSTORAGE_KEY}.${target}.by`) ?? '';
   const parts = preference.split('.');
   if (!parts[0] || !parts[1]) {
     return { sortBy: defaultSortBy, direction: defaultDirection };
   }
-  return { sortBy: parts[0], direction: parts[1] };
+  const sortBy = parts[0] as SortBy;
```

Comment:
Yeah, that's why I just left those as just strings.

Reply:
Yeah, but asserting after pulling from localstorage is expected 🤷